app/webpacker/controllers/terms_and_conditions_controller.js

```diff
@@ -1,12 +1,16 @@
 import ApplicationController from "./application_controller";
 
 export default class extends ApplicationController {
+  static targets = ["filename", "fileinput"];
   static values = {
     message: String,
   };
 
   connect() {
     super.connect();
+    this.fileinputTarget.addEventListener("change", (event) => {
+      this.filenameTarget.innerText = event.target.files[0].name;
```

Comment:
Each time the fileinput change, we update this little span that contains the file name: though we can simulate the fileinput behavior (ie. with the to-be-uploaded filename)